spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -852,6 +900,119 @@ def create_obj_in_recycled_slot(should_sample_original: false)
     end
   end
 
+  describe '#stats' do
+    it 'returns basic lifetime stats of stack recorder' do
+      num_serializations = 5
+
+      num_serializations.times do
+        stack_recorder.serialize
+      end
+
+      stats = stack_recorder.stats
+
+      expect(stats).to match(
+        hash_including(
+          :serialization_successes => num_serializations,
+          :serialization_failures => 0,
+
+          :serialization_time_ns_min => be > 0,
+          :serialization_time_ns_max => be > 0,
+          :serialization_time_ns_avg => be > 0,
+          :serialization_time_ns_total => be > 0,
+
+          :heap_recorder_snapshot => nil,
+        )
+      )
+
+      serialization_time_min, serialization_time_max, serialization_time_avg, serialization_time_total =
+        stats.values_at(
+          :serialization_time_ns_min,
+          :serialization_time_ns_max,
+          :serialization_time_ns_avg,
+          :serialization_time_ns_total
+        )
+
+      expect(serialization_time_min).to be <= serialization_time_avg
+      expect(serialization_time_avg).to be <= serialization_time_max
+      expect(serialization_time_total).to eq serialization_time_avg * num_serializations
```

Comment:
I wonder if this may cause some flakiness due to double rounding, e.g. it may be worth adding some `be_within(...)` here maybe?

Reply:
Good call!